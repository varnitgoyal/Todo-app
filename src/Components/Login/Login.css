.login {
    max-width: 40%;
    margin: auto;
    margin-top: 10%;
  }
  input[type="text"],
  input[type="password"] {
    width: 100%;
    padding: 12px 20px;
    margin: 8px 0;
    display: inline-block;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-sizing: border-box;
  }
  input[type="submit"] {
    width: 100%;
    background-color: #FFAA00;
    color: white;
    padding: 14px 20px;
    margin: 8px 0;
    border: none;
    border-radius: 4px;
    cursor: pointer;
  }
  