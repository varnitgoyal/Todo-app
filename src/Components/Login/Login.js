import React, { Component } from "react";
import "./Login.css";
import { Redirect } from "react-router-dom";
export class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      password: ""
    };
  }
  handleChange = event => {
    this.setState({
      ...this.state,
      [event.target.name]: event.target.value
    });
  };

  render() {
    const { isAuthenticated, handleSubmit } = this.props;

    if (isAuthenticated) {
      return <Redirect to="/todos" />;
    }

    return (
      <div>
        <form
          className="login"
          onSubmit={event => handleSubmit(event, this.state)}
        >
          <h2>Login</h2>
          <input
            type="text"
            onChange={this.handleChange}
            name="username"
            value={this.state.username}
            placeholder="username"
          />
          <input
            type="password"
            onChange={this.handleChange}
            name="password"
            value={this.state.password}
            placeholder="password"
          />
          <input type="submit" />
        </form>

        <p>username is admin</p>
        <p>password is admin</p>
      </div>
    );
  }
}

export default Login;
