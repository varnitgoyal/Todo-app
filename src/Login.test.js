import renderer from "react-test-renderer";
import Login from "./Components/Login/Login";
import React from "react";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallow,mount } from "enzyme";
import sinon from 'sinon';

Enzyme.configure({ adapter: new Adapter() });

describe("testing login component", () => {
  test("Is snapshot Same", () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("is login component rendered", () => {
    const wrapper = shallow(<Login />);
    expect(wrapper.find(".login_form").length).toBe(1);
  });

  test("does both text fields rendered",()=>{
      const wrapper=mount(<Login/>);
     
      expect(wrapper.containsMatchingElement(<input name="password"/>)).toEqual(true);
      expect(wrapper.containsMatchingElement(<input name="username"/>)).toEqual(true);

    
  })
  describe('Username input', () => {
  
    it('should respond to change event and change the state of the Login Component', () => {
     
     const wrapper = shallow(<Login />);
     wrapper.find('#username').simulate('change', {target: {name: 'username', value: 'blah@gmail.com'}});
     
    expect(wrapper.state('username')).toEqual('blah@gmail.com');
    })
   })
});
