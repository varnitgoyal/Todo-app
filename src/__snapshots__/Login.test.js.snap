// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing login component Is snapshot Same 1`] = `
<div
  className="login_form"
>
  <form
    className="login"
    onSubmit={[Function]}
  >
    <h2>
      Login
    </h2>
    <input
      id="username"
      name="username"
      onChange={[Function]}
      placeholder="username"
      type="text"
      value=""
    />
    <input
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value=""
    />
    <input
      type="submit"
    />
  </form>
  <p>
    username is admin
  </p>
  <p>
    password is admin
  </p>
</div>
`;
